atom/browser/atom_browser_client.cc

```diff
@@ -189,6 +209,28 @@ content::QuotaPermissionContext*
   return new AtomQuotaPermissionContext;
 }
 
+void AtomBrowserClient::SelectClientCertificate(
+    content::WebContents* web_contents,
+    net::SSLCertRequestInfo* cert_request_info,
+    scoped_ptr<content::ClientCertificateDelegate> delegate) {
+  auto command_line = base::CommandLine::ForCurrentProcess();
+  auto cert_path = command_line->GetSwitchValueNative(
+      switches::kClientCertificate);
+
+  // TODO(zcbenz): allow users to select certificate from
+  // client_cert list. Right now defaults to first certificate
+  // in the list.
+  scoped_refptr<net::X509Certificate> certificate;
+  if (cert_path.empty()) {
+    if (!cert_request_info->client_certs.empty())
+      certificate = cert_request_info->client_certs[0];
+  } else {
+    certificate = ImportCertFromFile(base::FilePath(cert_path));
+  }
+
+  delegate->ContinueWithCertificate(certificate.get());
```

Comment:
Calling `ContinueWithCertificate` with `nullptr` will finally reach here:
https://code.google.com/p/chromium/codesearch#chromium/src/net/url_request/url_request_job.cc&sq=package:chromium&type=cs&l=226

The code you were referencing checks for `handle` not `cert`.